docs/topics/dependencies.md

```diff
@@ -0,0 +1,123 @@
+
+- by creating a custom worker container image, which has your dependencies. This approach is [described in detail in this document](workers.md). In a nutshell, use this approach if you have the same dependency for multiple projects, or if your dependencies take a long time to pull.
+
+- without creating a custom container image:
+    - by adding a `brigade.json` file in your repository (similar to a `package.json` file) 
+that contains the dependencies and that are specific on every Brigade project.
+    - by directly using local dependencies located in your project repository.
+
+This document describes the last two approaches.
+
+> Here you can find a [repository that exemplifies both approaches here](https://github.com/radu-matei/brigade-javascript-deps).
+
+## Add custom dependencies using a `brigade.json` file
+
+If you need different dependencies for every Brigade project, this can be easily achieved 
+using a `brigade.json` file placed side-by-side the `brigade.js` file. This file contains 
+the dependency name and version, and has the following structure:
+
+```
+{
+    "dependencies": {
+        "is-thirteen": "2.0.0"
+    }
+}
+```
+Before starting to execute the `brigade.js` script, the worker will install the  
+dependencies using `yarn`, adding them to the `node_modules` folder.
+
+Then, in the `brigade.js` file, the new dependency can be used just like any 
+other NodeJS dependency:
+
+```
+const { events } = require("brigadier")
+const is = require("is-thirteen");
+
+events.on("exec", function (e, p) {
+    console.log("is 13 thirteen? " + is(13).thirteen());
+})
+```
+
+Now if we run a build for this project, we see the `is-thirteen` dependency added, 
+as well as the console log resulted from using it:
+
+```
+$ brig run brigade-86959b08a89af5b6b83f0ace6d9030f1fdca7ed8ea0a296e27d72e
+Event created. Waiting for worker pod named "brigade-worker-01cexrvrs08shcev26961cwd6n".
+Started build 01cexrvrs08shcev26961cwd6n as "brigade-worker-01cexrvrs08shcev26961cwd6n"
+installing is-thirteen@2.0.0
+prestart: src/brigade.js written
+[brigade] brigade-worker version: 0.14.0
+[brigade:k8s] Creating PVC named brigade-worker-01cexrvrs08shcev26961cwd6n
+is 13 thirteen? true
+[brigade:app] after: default event handler fired
+[brigade:app] beforeExit(2): destroying storage
+[brigade:k8s] Destroying PVC named brigade-worker-01cexrvrs08shcev26961cwd6n
+```
+
+Notes:
+
+- the dependencies should point the exact version (and not use the tilde ~ and caret ^ to indicate semver compatible versions)
+
+- when adding a custom dependency using `brigade.json`, `yarn` will add it side-by-side with [the worker's 
+dependencies](../../brigade-worker/package.json) - this means that the process will fail if a dependency that conflicts with one of the 
+worker's dependency is added. However, already existing dependencies (such as `@kubernetes/client-node`, `ulid` or `chai`) 
```

Comment:
typo: pluralization needed, should read `... worker's dependencies...`

Reply:
Curious -- what does the failure look like to the user?  Will it be obvious?

Reply:
Hm, that's a very good point - let me open a new issue for that. Thanks!